src/pub.c

```diff
@@ -576,10 +567,13 @@ static DPS_Status CallPubHandlers(DPS_Node* node, DPS_Publication* pub)
     /*
      * Iterate over the candidates and check that the pub strings are a match
      */
-    for (cdt = candidates; cdt; cdt = cdt->next) {
+    for (sub = node->subscriptions; sub != NULL; sub = sub->next) {
         int match;
-        ret = DPS_MatchTopicList(pub->shared->topics, pub->shared->numTopics, cdt->sub->topics,
-                                 cdt->sub->numTopics, node->separators, DPS_FALSE, &match);
+        if (!DPS_BitVectorIncludes(pub->shared->bf, sub->bf)) {
```

Comment:
Would be good to avoid calling DPS_BitVectorIncludes  again given it was called above. Most common case is going to be to zero or one match. Is there anything unique in the subscription that could be used to identify the matching candidate.